### PR TITLE
Merge pull request #1 from Blosc/master

### PR DESCRIPTION
**Pull request by oscargm98**
Pull request for fixing warnings and tests errors.

### CODE REVIEW

File: caterva/caterva_blosc.c [hunk 1]
```diff
@@ -315,7 +318,7 @@ int caterva_blosc_array_free(caterva_context_t *ctx, caterva_array_t **array) {
 }
 
 
-int caterva_blosc_array_repart_chunk(int8_t *rchunk, int rchunksize, void *chunk, int chunksize, caterva_array_t *array){
+int caterva_blosc_array_repart_chunk(int8_t *rchunk, int64_t rchunksize, void *chunk, int32_t chunksize, caterva_array_t *array){
```
Comment (FrancescAlted) [inline caterva/caterva_blosc.c:321]: Maybe `rchunksize` can be converted into a `int32_t` ?

Comment (oscargm98) [reply caterva/caterva_blosc.c:321]: Then we would have to add casts at lines 461, 464 and 588

Comment (FrancescAlted) [reply caterva/caterva_blosc.c:321]: Yeah, but sometimes practicality beats purity.  It is clear that `rchunksize` can never be larger than `int32_t` so I'd do the change and add the new three casts.  The fact that some API public calls have `size_t` arguments is a bit unfortunate, and we should ponder to always use `int32_t` for chunksizes/blocksizes.

Comment (oscargm98) [reply caterva/caterva_blosc.c:321]: it is OK then. I will do the change and I will also change the size_t type variables to int32_t variables. Should I open a new pull request for the new changes?

Comment (FrancescAlted) [reply caterva/caterva_blosc.c:321]: Yes please.  Thanks!

**Pull request merged**